src/daos-gen/daos-gen.c

```diff
@@ -0,0 +1,290 @@
+        daos_cont_info_t co_info = {0};
+#if DAOS_API_VERSION_MAJOR < 1
+        rc = daos_pool_connect(daos_args->src_pool_uuid, NULL, NULL, DAOS_PC_RW,
+                               &(daos_args->src_poh), &pool_info, NULL);
+#else
+        rc = daos_pool_connect(daos_args->src_pool_uuid, NULL, DAOS_PC_RW,
+                               &(daos_args->src_poh), &pool_info, NULL);
+#endif
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to connect to pool: "DF_RC, DP_RC(rc));
+            return 1;
+        }
+
+        /* create cont and open */
+        rc = daos_cont_create(daos_args->src_poh, daos_args->src_cont_uuid, NULL, NULL);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to create cont: "DF_RC, DP_RC(rc));
+            return 1;
+        }
+        rc = daos_cont_open(daos_args->src_poh, daos_args->src_cont_uuid,
+                            DAOS_COO_RW, &daos_args->src_coh, &co_info, NULL);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to open container: "DF_RC, DP_RC(rc));
+            return 1;
+        }
+    }
+
+    /* broadcast rank 0's pool and cont handle to everyone else */
+    daos_bcast_handle(rank, &(daos_args->src_poh), &(daos_args->src_poh), POOL_HANDLE); 
+    daos_bcast_handle(rank, &(daos_args->src_coh), &(daos_args->src_poh), CONT_HANDLE); 
+
+    /* TODO: generate different types of data, and different for each key */
+	memset(buf, 'A', BUFLEN);
+    for (i = 0; i < num_objects; i++) {
+        oid[i] = dts_oid_gen(0);
+                                   
+        rc = daos_obj_generate_oid(daos_args->src_coh, &oid[i], ofeats, OC_RP_XSF, 0, 0);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to generate oid: "DF_RC, DP_RC(rc));
+            return 1;
+        }
+
+    	/** create the KV store */
+        rc = daos_kv_open(daos_args->src_coh, oid[i], DAOS_OO_RW, &oh[i], NULL);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to open kv object: "DF_RC, DP_RC(rc));
+            return 1;
+        }
+
+        /* insert keys */
+        for (j = 0; j < keys_per_object; j++) {
+            sprintf(key, key_fmt, j);
+            rc = daos_kv_put(oh[i], DAOS_TX_NONE, 0, key, BUFLEN, buf, NULL);
+            if (rc != 0) {
+                MFU_LOG(MFU_LOG_ERR, "Failed to put kv object: "DF_RC, DP_RC(rc));
+                return 1;
+            }
+        }
+
+        rc = daos_kv_close(oh[i], NULL);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "Failed to close kv object: "DF_RC, DP_RC(rc));
+            return 1;
+        }
+    }
+
+    /* calculate total objects and num_keys_per object, each rank
+     * generates same amount */
+    if (rank == 0) {
+        int total_num_objects = size * num_objects; 
+        uuid_unparse(daos_args->src_cont_uuid, uuid_str);
+        printf("Container UUID: %s\n\ttotal objects:%d\n"
+               "\tkeys per object:%d\n", uuid_str, total_num_objects, keys_per_object);
+    }
+
```

Comment:
@dsikich I'd recommend putting a `label` here so error conditions above can `goto` for proper cleanup.